### ISSUE TITLE
What is the difference between this package and gplearn?

### ISSUE THREAD

**Issue by hengzhe-zhang** [marcovirgolin/genepro#2]
Hello! Thanks for providing such a wonderful tool. I've been following this project for a long time. It seems that this project is quite similar to gplearn. Is there any difference between this package and gplearn? By the way, will GP-GOMEA be integrated in this package? I am just curious about these things, and reinventing the wheel is definitely not a bad thing.

**Comment by marcovirgolin** [comment]
Hi, thanks for the interest!

I made this package to use it to teach part of a master course this year.
To be able to teach GP and prepare assignments well, I want to know the code base I am using very well.
Plus, I want the code base to be minimal.

I am not sure about the difference with `gplearn`. From a quick look at it, it seems to me that `gplearn` is a much more mature and popular project than this one. But again, this project is very much for my personal use to teach, and I made it public just in case anyone may find it interesting/useful.

I do not plan to integrate GP-GOMEA here, at least for now, for two reasons:
- I want to keep this project minimal, because it makes it easier for students to go through the code and learn what is going on. 
- In my experience, python is just too slow to run GP with decently large population sizes (e.g., at least thousands of solutions) in a reasonable time. GP-GOMEA computes statistics on the population to decide how to perform crossover, and therefore the population needs to be sufficiently large for these statistics to be accurate and useful. So, GP-GOMEA would probably be too slow to be usable if implemented in python (unless perhaps one starts to use stuff like cython or numba)
